### PR TITLE
Add notebook documentation

### PR DESCRIPTION
**Pull request by Paul-Saves**

### CODE REVIEW

**Review by relf — CHANGES_REQUESTED** [review]
Nitpicking: Would you mind renaming in SMT_MixedInteger_application.ipynb to have some kind of naming consistency?

**Comment by Paul-Saves** [comment]
See Slack. MGP model option has not be enable in the applications and decorators.

File: smt/applications/ego.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 from smt.applications.mixed_integer import MixedIntegerContext
 from smt.utils.misc import compute_rms_error
 
-from smt.surrogate_models import KPLS, KRG, KPLSK
+from smt.surrogate_models import KPLS, KRG, KPLSK,MGP
```
Comment (relf) [inline smt/applications/ego.py:20]: Nitpicking: would you mind passing black or adding a space after the comma?

File: smt/applications/ego.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def _initialize(self):
         declare(
             "surrogate",
             KRG(print_global=False),
-            types=(KRG, KPLS, KPLSK),
+            types=(KRG, KPLS, KPLSK,MGP),
```
Comment (relf) [inline smt/applications/ego.py:110]: same as previous

**Review by relf — CHANGES_REQUESTED** [review]

File: smt/applications/ego.py (hunk 1 above)
Comment (Paul-Saves) [reply smt/applications/ego.py]: I applied black on the complete folder so other files as marked as changed

Comment (relf) [reply smt/applications/ego.py]: Yes, I've seen that. It is ok.

**Review by relf — APPROVED** [review]

**Pull request merged**